matrix_client/room.py

```diff
@@ -624,6 +625,24 @@ def modify_required_power_levels(self, events=None, **kwargs):
         except MatrixRequestError:
             return False
 
+    def set_invite_only(self, invite_only):
```

Comment:
I really think we should make this take a (fake) enum rather than a boolean
(change method name also obviously). That way if in the future `knock` or
`private` are used, we can accommodate that without changing the API for this
class. Thoughts?

Reply:
I wanted to have something intuitive. As I understood it, the goal of those higher-level methods is to ease the use of the CS API, hence it didn't make sense to me to have basically the same method in `room.py` and `api.py`.
Also, I'm thinking that if `knock` or `private` are used, they will necessitate changes to this method anyway, so it may be good to wait to include them. And depending on what behavior they introduce, maybe it will make sense to have two different setters for the same underlying api call!

Reply:
You convinced me. Thanks again!